packages/grid/x-data-grid-premium/src/hooks/features/export/serializer/excelSerializer.ts

```diff
@@ -0,0 +1,235 @@
+import type * as Excel from 'exceljs';
+import {
+  GridStateColDef,
+  GridRowId,
+  GridColDef,
+  GridValueFormatterParams,
+  GridApi,
+  ValueOptions,
+} from '@mui/x-data-grid-pro';
+import { buildWarning } from '@mui/x-data-grid/internals';
+import { GridExceljsProcessInput } from '../gridExcelExportInterface';
+
+const getExcelJs = () => import('exceljs');
```

Comment:
I will be curious to see how this pattern behaves in production. First time I see an attempt at it. I will be positively surprised if it works.